framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/builder/GenericScenarioBuilderImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.scenario.builder;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.kie.cloud.api.deployment.constants.DeploymentConstants;
+import org.kie.cloud.api.scenario.GenericScenario;
+import org.kie.cloud.api.scenario.builder.GenericScenarioBuilder;
+import org.kie.cloud.openshift.OpenShiftController;
+import org.kie.cloud.openshift.constants.OpenShiftConstants;
+import org.kie.cloud.openshift.constants.OpenShiftTemplateConstants;
+import org.kie.cloud.openshift.constants.ScenarioConstants;
+import org.kie.cloud.openshift.scenario.GenericScenarioImpl;
+
+public class GenericScenarioBuilderImpl implements GenericScenarioBuilder {
+
+    private OpenShiftController openshiftController;
+    private Map<String, String> envVariables;
```

Comment:
Something to think about for the future: These 2 maps illustrate it clearly that we're not taking advantage of static types: 
1) `envVariables` maps variable names to variable values
2) `kieAppTemplates` maps scenario names  to template urls. 

4 different kinds of entities, yet we're all representing them all with Strings.